torch/_subclasses/fake_tensor.py

```diff
@@ -422,11 +422,53 @@ def nyi(fake_mode, func, *args, **kwargs):
     assert func not in _device_not_kwarg_ops, f"NYI: {func}"
 
 
-# Meta tensors give you the ability to run PyTorch code without having to
-# actually do computation through tensors allocated on a `meta` device.
-# Because the device is `meta`, meta tensors do not model device propagation.
-# FakeTensor extends MetaTensors to also carry an additional `fake_device`
-# which tracks devices that would have been used.
+@register_op_impl(
+    lambda func: func in (aten.convolution.default, aten.convolution_backward.default)
+)
+def conv(fake_mode, func, *args, **kwargs):
+    args, kwargs = normalize_function(
+        func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+    )
+    device = kwargs["input"].fake_device
+    # need to re-enable mode so the tensors report fake device
+    with fake_mode:
+        if func is aten.convolution.default:
+            conv_backend = torch._C._select_conv_backend(*args, **kwargs)
+        else:
+            conv_backend = torch._C._select_conv_backend(
+                kwargs["input"],
+                kwargs["weight"],
+                bias=None,
+                stride=kwargs["stride"],
+                padding=kwargs["padding"],
+                dilation=kwargs["dilation"],
+                transposed=kwargs["transposed"],
+                output_padding=kwargs["output_padding"],
+                groups=kwargs["groups"],
+                bias_sizes=kwargs["bias_sizes"],
+            )
+        mem_fmt = torch._C._conv_determine_backend_memory_format(
+            kwargs["input"], kwargs["weight"], conv_backend
+        )
```

Comment:
You don't actually care about the backend, right? Just the output memory format. Maybe only expose that?

Reply:
yea, `_select_conv_backend` was already exposed fwiw